### DISCUSSION TITLE
UI cannot display tables from non-public schema

### DISCUSSION THREAD

**Discussion by shorn** [supabase/supabase#549] [Questions]
I have a table in a newly created `private` schema.

Created with the SQL:
```
create schema private;

-- drop table private.cabbage_error;

create table private.cabbage_error (
  id integer generated by default as identity
    constraint cabbage_error_pkey primary key,
  content json not null
);

grant insert, select, update, delete, truncate, references, trigger on private.cabbage_error to supabase_auth_admin;
```

The table is visible in the list of tables in the table editor, correctly listed as being in the `private` schema:
![image](https://user-images.githubusercontent.com/216456/104832887-6a0baa00-58e0-11eb-81a1-dcaf5a0f295c.png)

But if you try to view the content of the table, the UI shows the following error:
![image](https://user-images.githubusercontent.com/216456/104832890-6d9f3100-58e0-11eb-873c-610553a5f9fe.png)

The error message implying that it's looking in the `public` schema.

**Comment by kiwicopple** [comment]
Sorry about this one @shorn - we are still waiting for the updates in PostgREST to support multiple schemas - Steve has actually done the work, so it shouldn't be far away 👍

**Reply by steve-chavez** [reply]
@shorn Since you called your schema `private`, I guess you don't want to expose it through the API right?

**Reply by shorn** [reply]
> 
> 
> @shorn Since you called your schema `private`, I guess you don't want to expose it through the API right?

Yup, as per this discussion about about [stored procedures](https://github.com/supabase/supabase/discussions/547). 
I intend for this table to be the data storage for my client side "error capturing" (using Sentry.js client and hacking the transport to deliver to Supabase) and I'd want that data to be very private.

I understood one of the ways to do that with Supabase was to put the table in a separate schema, then define a function in the public schema to act as a write-only API.

I understand I can probably do this with RLS policies too (as per [this topic](https://github.com/supabase/supabase/discussions/372)) 

I'm just trying out all these different ways to do this stuff with Supabase to see what works best for me.

Please note: this stuff is just me playing around trying to learn about Supabase, and maybe write an article or two about what I learned - I'm not building a real product right now.
